File: javascript/src/sliding-window/max-consecutive-ones.js
import { parseAssignments } from "../shared/io.js";
export const implemented = true;
export function solve(input) {
  const { nums, k } = parseAssignments(input);
  let left = 0, zeros = 0, best = 0;
  for (let right = 0; right < nums.length; right++) {
    if (nums[right] === 0) zeros++;
    while (zeros > k) {
      if (nums[left] === 0) zeros--;
      left++;
    }
    best = Math.max(best, right - left + 1);
  }
  return best;
}

File: python/solutions/sliding_window/max_consecutive_ones.py
from typing import List

class Solution:
    def longestOnes(self, nums: List[int], k: int):
        left = 0
        zeros = 0
        best = 0

        for right, value in enumerate(nums):
            if value == 0:
                zeros += 1
            while zeros > k:
                if nums[left] == 0:
                    zeros -= 1
                left += 1
            best = max(best, right - left + 1)

        return best
